test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,579 @@
+
+	By("wait until bastion is deleted")
+	err := extensions.WaitUntilExtensionCRDeleted(
+		ctx,
+		c,
+		logger,
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Bastion{} },
+		extensionsv1alpha1.BastionResource,
+		bastion.Namespace,
+		bastion.Name,
+		10*time.Second,
+		16*time.Minute,
+	)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func newCluster(name string) (*extensionsv1alpha1.Cluster, *controller.Cluster) {
+	providerConfig := &awsv1alpha1.CloudProfileConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "aws.provider.extensions.gardener.cloud/v1alpha1",
+			Kind:       "CloudProfileConfig",
+		},
+		MachineImages: []awsv1alpha1.MachineImages{
+			{
+				Name: "ubuntu",
+				Versions: []awsv1alpha1.MachineImageVersion{
+					{
+						Version: "20.04.20210223",
+						Regions: []awsv1alpha1.RegionAMIMapping{
+							{
+								Name: *region,
+								AMI:  "ami-08bac620dc84221eb",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	shoot := &v1beta1.Shoot{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "core.gardener.cloud/v1beta1",
+			Kind:       "Shoot",
+		},
+		Spec: v1beta1.ShootSpec{
+			Region: *region,
+		},
+	}
+
+	providerConfigJSON, _ := json.Marshal(providerConfig)
+	shootJSON, _ := json.Marshal(shoot)
+
+	extensionscluster := &extensionsv1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: extensionsv1alpha1.ClusterSpec{
+			CloudProfile: runtime.RawExtension{
+				Object: &core.CloudProfile{
+					Spec: core.CloudProfileSpec{
+						ProviderConfig: &runtime.RawExtension{
+							Object: providerConfig,
+						},
+					},
+				},
+			},
+			Shoot: runtime.RawExtension{
+				Object: shoot,
+				Raw:    shootJSON,
+			},
+		},
+	}
+
+	corecluster := &controller.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		CloudProfile: &v1beta1.CloudProfile{
+			Spec: v1beta1.CloudProfileSpec{
+				ProviderConfig: &runtime.RawExtension{
+					Object: providerConfig,
+					Raw:    providerConfigJSON,
+				},
+			},
+		},
+		Shoot: shoot,
+	}
+
+	return extensionscluster, corecluster
+}
+
+func newBastion(namespace string) (*extensionsv1alpha1.Bastion, error) {
+	const sshPublicKey = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDcSZKq0lM9w+ElLp9I9jFvqEFbOV1+iOBX7WEe66GvPLOWl9ul03ecjhOf06+FhPsWFac1yaxo2xj+SJ+FVZ3DdSn4fjTpS9NGyQVPInSZveetRw0TV0rbYCFBTJuVqUFu6yPEgdcWq8dlUjLqnRNwlelHRcJeBfACBZDLNSxjj0oUz7ANRNCEne1ecySwuJUAz3IlNLPXFexRT0alV7Nl9hmJke3dD73nbeGbQtwvtu8GNFEoO4Eu3xOCKsLw6ILLo4FBiFcYQOZqvYZgCb4ncKM52bnABagG54upgBMZBRzOJvWp0ol+jK3Em7Vb6ufDTTVNiQY78U6BAlNZ8Xg+LUVeyk1C6vWjzAQf02eRvMdfnRCFvmwUpzbHWaVMsQm8gf3AgnTUuDR0ev1nQH/5892wZA86uLYW/wLiiSbvQsqtY1jSn9BAGFGdhXgWLAkGsd/E1vOT+vDcor6/6KjHBm0rG697A3TDBRkbXQ/1oFxcM9m17RteCaXuTiAYWMqGKDoJvTMDc4L+Uvy544pEfbOH39zfkIYE76WLAFPFsUWX6lXFjQrX3O7vEV73bCHoJnwzaNd03PSdJOw+LCzrTmxVezwli3F9wUDiBRB0HkQxIXQmncc1HSecCKALkogIK+1e1OumoWh6gPdkF4PlTMUxRitrwPWSaiUIlPfCpQ== your_email@example.com"
```

Comment:
who has the private key? :D
As alternative we could also generate a new one for each test. WDYT?

Reply:
It's the same key that is used in the infrastructure tests: https://github.com/gardener/gardener-extension-provider-aws/blob/master/test/integration/infrastructure/infrastructure_test.go#L424

Reply:
seems like @ialidzhikov introduced this

@ialidzhikov do you see a (security) issue that this public key is used for the bastion test?

Reply:
Why should this be an issue? It's (1) a public key which is (2) only used in a test. Let's keep it.

Reply:
we have test-private-keys all over the place so I wanted to be sure that this private key is not stored anywhere